library/src/main/java/com/bumptech/glide/load/resource/bitmap/Downsampler.java

```diff
@@ -248,17 +248,6 @@ static void calculateScaling(DownsampleStrategy downsampleStrategy, int degreesT
     float adjustedScaleFactor = powerOfTwoSampleSize * exactScaleFactor;
 
     options.inSampleSize = powerOfTwoSampleSize;
-    // Density scaling is only supported if inBitmap is null prior to KitKat. Avoid setting
-    // densities here so we calculate the final Bitmap size correctly.
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```

Comment:
This is the first part of the fix, it seems that the values here are part of the problem